terraform/node_resource_plan_instance.go

```diff
@@ -128,11 +128,12 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 
 	// Refresh, maybe
 	if !n.skipRefresh {
-		instanceRefreshState, refreshDiags := n.refresh(ctx, instanceRefreshState)
+		s, refreshDiags := n.refresh(ctx, instanceRefreshState)
 		diags = diags.Append(refreshDiags)
 		if diags.HasErrors() {
 			return diags
 		}
+		instanceRefreshState = s
```

Comment:
Is the temporary variable `s` needed here? I don't see the logical difference between this and the prior code, given the shortcut return on error.

Reply:
`instanceRefreshState` is passed to `plan` below, but `instanceRefreshState` here is scoped only to this block and shadows the outer variable. The temporary value ensures both that we don't shadow the outer variable, and that the new value is re-assigned.

Reply:
Ah, I see now, thanks for clarifying!